Src/FluentAssertions/Specialized/ExceptionAssertions.cs

```diff
@@ -191,6 +192,54 @@ public ExceptionAssertions(IEnumerable<TException> exceptions)
             return this;
         }
 
+        private IEnumerable<Exception> GetExpectedInnerExceptions(Type innerException, string because = null,
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(innerException, nameof(innerException));
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected inner {0}{reason}, but ", innerException)
+                .ForCondition(Subject is not null)
+                .FailWith("no exception was thrown.")
+                .Then
+                .ForCondition(Subject.Any(e => e.InnerException is not null))
+                .FailWith("the thrown exception has no inner exception.")
+                .Then
+                .ClearExpectation();
+
+            Exception[] expectedInnerExceptions = Subject
+                .Select(e => e.InnerException)
+                .Where(e => e?.GetType() == innerException)
+                .ToArray();
+
+            Execute.Assertion
+                .ForCondition(expectedInnerExceptions.Any())
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected inner {0}{reason}, but found {1}.", innerException, SingleSubject.InnerException);
+
+            return expectedInnerExceptions;
+        }
+
+        private IEnumerable<Exception> GetInnerExceptionExactly(Type innerException, string because = null,
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(innerException, nameof(innerException));
+
+            WithInnerException(innerException, because, becauseArgs);
```

Comment:
🔧 Instead of creating a seemingly recursive call (which it didn't after digging through the code), I would just call the `GetExpectedInnerExceptions` method (and reverse their order in the file for readability).